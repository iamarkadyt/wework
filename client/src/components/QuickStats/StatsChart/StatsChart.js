import React, { Fragment } from 'react'
import {
    FaThumbsUp as IcoLike,
    FaComments as IcoComments
} from 'react-icons/fa'

const StatsChart = ({
    authedUser: {
        name, avatar
    },
    usersProfile: {
        title, company, status
    },
    stats: {
        followers: followersCount,
        following: subscriptionsCount,
        postCount,
        totalLikes,
        totalComments
    }
}) => (
        <Fragment>
            <section className="QuickStats-name">
                <img src={avatar} alt=""></img>
                <h2>{name}</h2>
                <p>{title}{company && ` at ${company}`}</p>
                <p>{status}</p>
            </section>
            <section className="QuickStats-rows">
                <h3>Your stats:</h3>
                <div>
                    <span>Following:</span>
                    <div />
                    <span>{subscriptionsCount}</span>
                </div>
                <div>
                    <span>Your followers:</span>
                    <div />
                    <span>{followersCount}</span>
                </div>
                <div>
                    <span>Total posts:</span>
                    <div />
                    <span>{postCount}</span>
                </div>
            </section>
            <section className="QuickStats-likes-comments">
                <h3>Total likes and comments on your posts:</h3>
                <div>
                    <p><IcoLike /> {totalLikes}</p>
                    <p><IcoComments /> {totalComments}</p>
                </div>
            </section>
        </Fragment>
    )

export default StatsChart
