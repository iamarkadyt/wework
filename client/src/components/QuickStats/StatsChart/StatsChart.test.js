import React from 'react'
import { shallow } from 'enzyme'
import StatsChart from './StatsChart'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    authedUser: {
      name: 'Mike',
      avatar: undefined
    },
    usersProfile: {
      title: 'Construction Worker',
      company: 'Construction Workers United',
      status: 'Not Open For Employment'
    },
    stats: {
      followers: 1038,
      following: 1270,
      postCount: 20,
      totalLikes: 503,
      totalComments: 121
    }
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<StatsChart {...props} />)
  }
  return mountedComponent
}

describe('StatsChart', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})


