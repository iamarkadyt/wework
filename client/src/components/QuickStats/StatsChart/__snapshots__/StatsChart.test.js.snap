// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsChart matches snapshot 1`] = `
<Fragment>
  <section
    className="QuickStats-name"
  >
    <img
      alt=""
    />
    <h2>
      Mike
    </h2>
    <p>
      Construction Worker
       at Construction Workers United
    </p>
    <p>
      Not Open For Employment
    </p>
  </section>
  <section
    className="QuickStats-rows"
  >
    <h3>
      Your stats:
    </h3>
    <div>
      <span>
        Following:
      </span>
      <div />
      <span>
        1270
      </span>
    </div>
    <div>
      <span>
        Your followers:
      </span>
      <div />
      <span>
        1038
      </span>
    </div>
    <div>
      <span>
        Total posts:
      </span>
      <div />
      <span>
        20
      </span>
    </div>
  </section>
  <section
    className="QuickStats-likes-comments"
  >
    <h3>
      Total likes and comments on your posts:
    </h3>
    <div>
      <p>
        <FaThumbsUp />
         
        503
      </p>
      <p>
        <FaComments />
         
        121
      </p>
    </div>
  </section>
</Fragment>
`;
